iModelJsNodeAddon/api_package/ts/src/test/DgnDb.test.ts

```diff
@@ -31,6 +31,44 @@ describe("basic tests", () => {
     done();
   });
 
+  it("compress/decompress", () => {
+    const assertCompressAndThenDecompress = (sourceData: Uint8Array) => {
+      const compressData = iModelJsNative.DgnDb.zlibCompress(sourceData);
+      const decompressData = iModelJsNative.DgnDb.zlibDecompress(compressData, sourceData.length);
+      assert.deepEqual(sourceData, decompressData);
+    };
+    const randomData = (n: number) => {
+      return Uint8Array.from(Array.from({ length: n }, () => Math.floor(Math.random() * 10)));
+    };
+    assertCompressAndThenDecompress(randomData(1024 * 1));
+    assertCompressAndThenDecompress(randomData(1024 * 2));
+    assertCompressAndThenDecompress(randomData(1024 * 4));
+    assertCompressAndThenDecompress(randomData(1024 * 8));
+    assertCompressAndThenDecompress(randomData(1024 * 16));
+
+    const seedUri = path.join(getOutputDir(), "compress-decompress.bim");
+    if (fs.existsSync(seedUri)) {
+      fs.unlinkSync(seedUri);
+    }
+    const iModelDb = new iModelJsNative.DgnDb();
+    iModelDb.createIModel(seedUri, { rootSubject: { name: "test file" } });
+
+    const propData = randomData(1024 * 32);
+    iModelDb.saveFileProperty({ namespace: "test", name: "test" }, "hello", propData);
+    iModelDb.saveChanges();
+    const stmt = new iModelJsNative.SqliteStatement();
+    stmt.prepare(iModelDb, "SELECT [RawSize], [Data] FROM [be_Prop] WHERE [NameSpace] = 'test' AND [Name] = 'test'");
+    assert.equal(stmt.step(), DbResult.BE_SQLITE_ROW);
+    const rawSize = stmt.getValueInteger(0);
+    assert.notEqual(rawSize, 0);
+    const blob = stmt.getValueBlob(1);
+    stmt.dispose();
+    const propDecompressData = iModelJsNative.DgnDb.zlibDecompress(blob, rawSize);

```

Comment:
Based off of this test, I'm assuming that saveFileProperty itself compresses the propData passed to it? Is that correct? How would a user know that this is the case? Feels like we should document that on saveFileProperty in itwinjs-core even if it is internal.

Reply:
Yes, and it's conditionally compressed by checking if compression increases the size of the buffer. If it does compress it sets another property called rawSize.

The addon method I added is not supposed to be called by the user. It's meant to be used by internal code in the conflict handler to record the result before/after value for conflict and its resolution in be_prop table. At the end of pull merge application can enumerate what the conflicts were detected and what was the decision taken by default and that app can then decide to do nothing or override the default resolution picking one of the old/new values and calling db.SaveFileProperty(). Note be_prop has no lock so two user can change it at the same time.